src/main/java/org/apache/jackrabbit/filevault/maven/packaging/GenerateMetadataMojo.java

```diff
@@ -852,7 +868,7 @@ private Properties computeProperties(String dependenciesString) {
 
                 // todo: add support for patterns
                 if (destFileName == null) {
-                    destFileName = source.getName();
+                    destFileName = Text.getName(embedArtifactLayout.pathOf(artifact));
```

Comment:
This will always assume that the artifactId together with (optional) classifier, version and extension is being used as the filename, but in some cases this is being overwritten with the `finalName` element within `build` (https://maven.apache.org/pom.html#BaseBuild_Element). I am currently looking for a simpler API replacement which considers this as well and only returns the filename but not the path.

Reply:
I think in case source is a directory we should instead generate an artificial name on our own to make it clear that there is no file yet, also there should be at least a DEBUG log statement in case this is generated. Something like "OtherModulesNonExistingBundle_<OtherMavenProject.build.finalName>"

Reply:
But in that case, wouldn't it just be easier to take `finalName` into account? We can search for the project producing the artifact just as easily here, and take the finalName into consideration without too much trouble, and that would result in a more stable filter.xml generation, for whatever tooling depends on it.

Reply:
Ok, let us just take the final name here. Only the classifier and extension is not yet added there (compare with https://github.com/apache/maven-jar-plugin/blob/78dbb2918fcb2086fe5601eb95bb294bcc274639/src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java#L177).

Reply:
Updated. Please review https://github.com/apache/jackrabbit-filevault-package-maven-plugin/pull/22/files#diff-ef7fb30adc4d3d5e46cd2c7814d9b3afR871